flang/include/flang/Parser/provenance.h

```diff
@@ -229,7 +233,8 @@ class CookedSource {
   void set_number(int n) { number_ = n; }
 
   CharBlock AsCharBlock() const { return CharBlock{data_}; }
-  std::optional<ProvenanceRange> GetProvenanceRange(CharBlock) const;
+  std::optional<ProvenanceRange> GetProvenanceRange(
```

Comment:
Would it be possible & cleaner to add an `allSources_` member to `CookedSource`?

Reply:
Looking at the lifetime of the objects AllSources lifetime > AllCookedSource lifetime > CookedSource lifetime, and the fact that the pointer duplication should be acceptable (not thousands of cooked source expected in normal Fortran programs), this seems possible and cleaner to me.

Done.